lib/network/http2_network_config.go

```diff
@@ -0,0 +1,138 @@
+package sebaknetwork
+
+import (
+	"errors"
+	"io"
+	"os"
+	"time"
+
+	"boscoin.io/sebak/lib/common"
+)
+
+type TLSInfo interface {
+	GetTLSCertInfo(*sebakcommon.Endpoint) (string, error)
+	GetTLSKeyInfo(*sebakcommon.Endpoint) (string, error)
+}
+
+type ValidTLSInfo struct{}
+
+func (t ValidTLSInfo) GetTLSCertInfo(endpoint *sebakcommon.Endpoint) (result string, err error) {
+	query := endpoint.Query()
+	if v := query.Get("TLSCertFile"); len(v) < 1 {
+		err = errors.New("'TLSCertFile' is missing")
+		return
+	} else {
+		result = v
+	}
+	return
+}
+
+func (t ValidTLSInfo) GetTLSKeyInfo(endpoint *sebakcommon.Endpoint) (result string, err error) {
+	query := endpoint.Query()
+	if v := query.Get("TLSKeyFile"); len(v) < 1 {
+		err = errors.New("'TLSKeyFile' is missing")
+		return
+	} else {
+		result = v
+	}
+	return
+}
+
+type HTTP2NetworkConfig struct {
+	NodeName string
+	Addr     string
+
+	ReadTimeout,
+	ReadHeaderTimeout,
+	WriteTimeout,
+	IdleTimeout time.Duration
+
+	TLSCertFile,
+	TLSKeyFile string
+
+	HTTP2LogOutput io.Writer
+}
+
+func NewHTTP2NetworkConfigFromEndpoint(tls TLSInfo, endpoint *sebakcommon.Endpoint) (config HTTP2NetworkConfig, err error) {
```

Comment:
Setting TLS certificate files by `TLSInfo` looks nice. If it is used for testing mainly, I think there are another the easier way. Instead of `TLSInfo`, we can set the `TLSCertFile` and `TLSKeyFile` at the 'endpoint' directly in testing.